docs/source/cuda/memory.rst

```diff
@@ -110,3 +110,28 @@ unlike traditional dynamic memory management.
    The array is private to the current thread.  An array-like object is
    returned which can be read and written to like any standard array
    (e.g. through indexing).
+
+SmartArray
```

Comment:
I would put `(experimental)` in the section heading and take it out of the text below.  I think it will make it easier to spot.